docs/examples/paddle/paddle-external_input.ipynb

```diff
@@ -0,0 +1,215 @@
+    "            self.i = (self.i + 1) % self.n\n",
+    "        return (batch, labels)\n",
+    "\n",
+    "    @property\n",
+    "    def size(self,):\n",
+    "        return self.data_set_len\n",
+    "\n",
+    "    next = __next__\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Defining the pipeline\n",
+    "\n",
+    "Now the pipeline itself will be defined. First of all, a framework iterator will be used so we need to make sure that images and the output of the pipeline are uniforms in size, so resize operator is used. Also, `iter_setup` will raise the StopIteration exception when the AdvancedExternalInputIterator run of data. Worth notice is that iterator needs to be recreated so next time `iter_setup` is called it has ready data to consume."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class ExternalSourcePipeline(Pipeline):\n",
+    "    def __init__(self, batch_size, num_threads, device_id, external_data):\n",
+    "        super(ExternalSourcePipeline, self).__init__(batch_size,\n",
+    "                                      num_threads,\n",
+    "                                      device_id,\n",
+    "                                      seed=12)\n",
+    "        self.input = ops.ExternalSource()\n",
+    "        self.input_label = ops.ExternalSource()\n",
+    "        self.decode = ops.ImageDecoder(device = \"mixed\", output_type = types.RGB)\n",
+    "        self.res = ops.Resize(device=\"gpu\", resize_x=240, resize_y=240)\n",
+    "        self.cast = ops.Cast(device = \"gpu\",\n",
+    "                             dtype = types.UINT8)\n",
+    "        self.external_data = external_data\n",
+    "        self.iterator = iter(self.external_data)\n",
+    "\n",
+    "    def define_graph(self):\n",
+    "        self.jpegs = self.input()\n",
+    "        self.labels = self.input_label()\n",
+    "        images = self.decode(self.jpegs)\n",
+    "        images = self.res(images)\n",
+    "        output = self.cast(images)\n",
+    "        return (output, self.labels)\n",
+    "\n",
+    "    def iter_setup(self):\n",
+    "        try:\n",
+    "            (images, labels) = self.iterator.next()\n",
+    "            self.feed_input(self.jpegs, images)\n",
+    "            self.feed_input(self.labels, labels)\n",
+    "        except StopIteration:\n",
+    "            self.iterator = iter(self.external_data)\n",
+    "            raise StopIteration"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Using the pipeline\n",
+    "\n",
+    "At the end let us see how it works. Please also notice the usage of `last_batch_padded` that tell iterator that the difference between data set size and batch size alignment is padded by real data that could be skipped at when provided to the framework (`fill_last_batch`):"
```

Comment:
I don't know if it is worth to mention here that the dataset used here has 21 samples, so it is divisible without any reminder by the batch size.

Reply:
To be honest, I just copy pasted those from other plugins.